bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -104,4 +159,107 @@ public static String getBroadcastAddress() {
         }
     }
 
+    /**
+     * Converts a netmask in bits into a string representation
+     * i.e. 24 bits -> 255.255.255.0
+     *
+     * @param prefixLength bits of the netmask
+     * @return string representation of netmask (i.e. 255.255.255.0)
+     */
+    public static @NonNull String networkPrefixLengthToNetmask(int prefixLength) {
+        if (prefixLength > 31 || prefixLength < 1) {
+            throw new IllegalArgumentException("Network prefix length is not within bounds");
+        }
+
+        int ipv4Netmask = 0xFFFFFFFF;
+        ipv4Netmask <<= (32 - prefixLength);
+
+        byte[] octets = new byte[] { (byte) (ipv4Netmask >>> 24), (byte) (ipv4Netmask >>> 16),
+                (byte) (ipv4Netmask >>> 8), (byte) ipv4Netmask };
+
+        String result = "";
+        for (int i = 0; i < 4; i++) {
+            result += octets[i] & 0xff;
+            if (i < 3) {
+                result += ".";
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Get the network address a specific ip address is in
+     *
+     * @param ipAddressString ipv4 address of the device (i.e. 192.168.5.1)
+     * @param netMask netmask in bits (i.e. 24)
+     * @return network a device is in (i.e. 192.168.5.0)
+     */
+    public static @NonNull String getIpv4NetAddress(@NonNull String ipAddressString, short netMask) {
```

Comment:
If you want to assure to always return valid non-null results, you should at least also throw IllegalArgumentExceptions. Or what would you return if I pass you "ABC" as my ipAddressString?